source/extensions/filters/common/ext_authz/ext_authz_impl.cc

```diff
@@ -174,11 +76,12 @@ void CheckRequestUtils::createHttpCheck(const Envoy::Http::StreamDecoderFilterCa
   Envoy::Http::StreamDecoderFilterCallbacks* cb =
       const_cast<Envoy::Http::StreamDecoderFilterCallbacks*>(callbacks);
 
-  const std::string service = getHeaderStr(headers.EnvoyDownstreamServiceCluster());
+  auto hdr = headers.EnvoyDownstreamServiceCluster();
+  std::string service = hdr ? std::string(hdr->value().getStringView()) : "";
```

Comment:
Why don't you use AttributeContextUtils::getHeaderStr here?

Reply:
it's private and seemed a weird thing to reach into the utils for.

Reply:
Just have `setSourcePeer` take a string view.